lib/mutant/expression/method.rb

```diff
@@ -3,44 +3,44 @@ class Expression
 
     # Explicit method expression
     class Method < self
+      include Anima.new(:scope_name, :scope_symbol, :method_name)
+      private(*anima.attribute_names)
 
       MATCHERS = IceNine.deep_freeze(
         '.' => Matcher::Methods::Singleton,
         '#' => Matcher::Methods::Instance
       )
+      private_constant(*constants(false))
 
-      register(
-        /\A(?<scope_name>#{SCOPE_PATTERN})(?<scope_symbol>[.#])(?<method_name>#{METHOD_NAME_PATTERN})\z/
-      )
+      REGEXP = /\A(?<scope_name>#{SCOPE_PATTERN})(?<scope_symbol>[.#])(?<method_name>#{METHOD_NAME_PATTERN})\z/.freeze
+
+      # Return syntax
+      #
+      # @return [String]
+      #
+      # @api private
+      #
+      def syntax
+        [scope_name, scope_symbol, method_name].join
+      end
+      memoize :syntax
 
       # Return method matcher
       #
       # @param [Env] env
       #
-      # @return [Matcher::Method]
+      # @return [Matcher]
       #
       # @api private
       #
       def matcher(env)
         methods_matcher = MATCHERS.fetch(scope_symbol).new(env, scope)
-        method = methods_matcher.methods.detect do |meth|
-          meth.name.equal?(method_name)
-        end or fail NameError, "Cannot find method #{method_name}"
-        methods_matcher.matcher.build(env, scope, method)
+
+        Matcher::Filter.new(methods_matcher, ->(subject) { subject.expression.eql?(self) })
```

Comment:
Would it make sense for `Matcher::Filter.new` to accept a block instead of the last argument being a proc?